PdCore/src/main/java/org/puredata/android/service/PdService.java

```diff
@@ -241,8 +241,14 @@ private Notification makeNotification(Intent intent, int icon, String title, Str
 				notificationManager.createNotificationChannel(channel);
 			}
 		}
-
-		PendingIntent pi = PendingIntent.getActivity(getApplicationContext(), 0, intent, 0);
+		
+            	PendingIntent pi = null;
+            	if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                	pi = PendingIntent.getActivity(getApplicationContext(), 0, intent, PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_IMMUTABLE);
+            	} else {
+                	pi = PendingIntent.getActivity(getApplicationContext(), 0, intent, PendingIntent.FLAG_UPDATE_CURRENT);
```

Comment:
I am against `FLAG_UPDATE_CURRENT` here. This may change the behaviour on older Android versions, and the existing code with using `0` was working in many apps.